docs/guides/streaming.md

```diff
@@ -0,0 +1,349 @@
+---
+title: Streaming
+description: When, why, and how to stream with React 18 and Remix's deferred API.
+---
+
+# Streaming
+
+Remix supports the [web streaming API](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API) as a first-class citizen. Additionally, JavaScript server runtimes have support for streaming responses to the client.
+
+One problem streaming in React 18 solves is for lazy-loading chunks of your code you don't want in your server render (for example, if a component doesn't support server-rendering). Once you enable React 18 streaming, you can use [`React.lazy`][react-lazy] on the server and the fallback will be rendered on the server and the client will fetch the code and render the components.
```

Comment:
@kentcdodds I don't believe this is 100% accurate. React.lazy *will* server render in React 18 when using one of the streaming renderer's instead of renderToString. We will be requiring one of those streaming renderers when using deferred, so I would re-word this or remove it.

Reply:
Wait. To be clear... If you're using `renderToString` (even with React 18) you can't use `React.lazy` right? That's what I'm trying to communicate here. That enabling streaming by using `renderToPipeableStream` is required if you want to use `React.lazy` on the server render. Am I mistaken?

Reply:
It's correct that `renderToString` will bail out on the server due to usage of streaming features (suspense) without using a streaming compatible rendering method . That is a byproduct of attempting to use streaming features in an invalid context and isn't really a "streaming feature", more of a byproduct of improper use and isn't behavior I would encourage relying on in general as as soon as you do actually start streaming that use-case breaks.